server/models/Comment.ts

```diff
@@ -0,0 +1,198 @@
+        model: 'Collectives',
+        key: 'id',
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: false,
+    },
+
+    FromCollectiveId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Collectives',
+        key: 'id',
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    },
+
+    CreatedByUserId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Users',
+        key: 'id',
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    },
+
+    ExpenseId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Expenses',
+        key: 'id',
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    },
+
+    UpdateId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Updates',
+        key: 'id',
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    },
+
+    ConversationId: {
+      type: DataTypes.INTEGER,
+      references: { key: 'id', model: 'Conversations' },
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    },
+
+    html: {
+      type: DataTypes.TEXT,
+      set(value: string) {
+        if (value) {
+          const cleanHtml = sanitizeHTML(value, sanitizeOptions).trim();
+          this.setDataValue('html', cleanHtml);
+        }
+      },
+    },
+
+    createdAt: {
+      type: DataTypes.DATE,
+      defaultValue: DataTypes.NOW,
+    },
+
+    updatedAt: {
+      type: DataTypes.DATE,
+      defaultValue: DataTypes.NOW,
+    },
+
+    deletedAt: {
+      type: DataTypes.DATE,
+    },
+  },
+  {
+    sequelize,
+    paranoid: true,
+    hooks: {
+      beforeCreate: instance => {
+        if (!instance.ExpenseId && !instance.UpdateId && !instance.ConversationId) {
+          throw new Error('Comment must be linked to an expense, an update or a conversation');
+        }
+      },
+      beforeDestroy: async (comment, options) => {
+        if (comment.ConversationId) {
+          const transaction = options.transaction;
+          const conversation = await models.Conversation.findOne({ where: { RootCommentId: comment.id }, transaction });
+          if (conversation) {
+            await conversation.destroy();
+            await models.Comment.destroy({
+              where: { id: { [Op.not]: comment.id }, ConversationId: conversation.id },
+              transaction,
+            });
+          }
+        }
+      },
+      afterUpdate: async (comment, options) => {
+        if (comment.ConversationId) {
+          const transaction = options.transaction;
+          const conversation = await models.Conversation.findOne({ where: { RootCommentId: comment.id }, transaction });
+          if (conversation) {
+            await conversation.update({ summary: comment.html }, { transaction });
+          }
+        }
+      },
```

Comment:
This is a real change compared to what we did before the migration: we're taking advantage of hooks rather than overriding the `destroy` method from sequelize (see https://github.com/opencollective/opencollective-api/pull/8255/files#diff-02b678ec46cee1ba66b4178ee4e435d7c0fd333d8245ba808c98e9cf86d19b37L150)